src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -311,7 +311,8 @@ public StackItem ToStackItem(ReferenceCounter referenceCounter)
                 SystemFee,
                 NetworkFee,
                 ValidUntilBlock,
-                Script,
+                new Array(Attributes.Select(p=> (StackItem)p.ToArray())),
```

Comment:
I don't think the attributes should be available in smart contracts.

Reply:
On some level maybe it's reasonable to provide attributes to sc developers for custom use. In some scenario, project parties can use different usage types to distinguish the tx type.

Reply:
But them are serialized. You can't deserialize them in a contract. So it's useless.